File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head.next == None:
            return None
        
        fast = head
        curr = head
        
        while(fast != None):
            if fast.next == None:
                break
            fast = fast.next
            fast = fast.next
            prev = curr
            curr = curr.next
        
        print(prev.val, curr.val)
        prev.next = curr.next
        return head
        
        
            
        
            
        
        

File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {
        ListNode* prev = nullptr;
        ListNode* curr = head;
        ListNode* fast = head;
            
        if(curr->next == nullptr){
            return nullptr;
        }
        
        while(fast != nullptr){
            if(fast->next == nullptr){
                break;
            }
            fast = fast->next;
            fast = fast->next;
            prev = curr;
            curr = curr->next;
        }
        
        prev->next = curr->next; 
        return head;
    }
};